astroquery/mast/cloud.py

```diff
@@ -154,8 +134,33 @@ def get_cloud_uri_list(self, data_products, include_bucket=True, full_url=False)
             List of URIs generated from the data products, list way contain entries that are None
             if data_products includes products not found in the cloud.
         """
+        s3_client = self.boto3.client('s3', config=self.config)
 
-        return [self.get_cloud_uri(product, include_bucket, full_url) for product in data_products]
+        paths = utils.mast_relative_path(data_products["dataURI"])
```

Comment:
This will break for GALEX data retrieval from the cloud, unfortunately. See here for how I tweaked this output in a recent PR following the new GALEX availability on the cloud: https://github.com/astropy/astroquery/blob/2b3d95498d0b6e73b9c3434ccc7d06cc50642f11/astroquery/mast/cloud.py#L119

Your two options would be to either: 
1. Make a tweak similar to this in `get_cloud_uri_list`
2. Make the tweak in `mast_relative_path` - this is probably the cleanest option. And then we can remove my tweak from `cloud.py`.

Reply:
re this suggestion: in the cloud WG we're working on generalizing these methods and move them out from mast. So a tweak at the place of usage, or a generically usable kwarg would be preferable rather than a hard-wired conditional on mast specific strings.

Reply:
What I did was fold in the `path.strip()` logic out of `cloud.py` and into `util.py`'s `mast_relative_path` function. In other words, I essentially went with option `2.` listed above. And because this is a MAST-specific function in `utils`, it makes sense for MAST-specific tweaks to take place in there. Let me know if that works for you!